core/src/main/scala/org/broadinstitute/dsde/rawls/dataaccess/HttpGoogleServicesDAO.scala

```diff
@@ -652,7 +646,11 @@ class HttpGoogleServicesDAO(
       }
 
       res <- allProcessedChunks.map(_.flatten).unsafeToFuture()
-    } yield res
+    } yield {
+        res collect {
```

Comment:
I spent a long time trying to figure out how to do this filtering above near line 643, where the `RawlsBillingAccount` is actually created. I was stumped (the Future seemed to complicate things), but would love to hear from Scala experts if there is a better way to do this.

Reply:
Apparently this can be further simplified to `res.filter(account => firecloudHasAccess.forall(access => access == account.firecloudHasAccess))`. The `forAll` returns true if the Option is empty.

(I personally find the `collect` the easiest to read/understand, but thankfully I have unit tests that explain the desired behavior)

Reply:
I believe you could use a collect up where the `RawlsBillingAccount` is actually created -- something like this might work

```
testDMBillingAccountAccess(acctName) collect {
            case firecloudHasAccount if (firecloudHasAccess.isEmpty || firecloudHasAccount == firecloudHasAccess.getOrElse(true)) => RawlsBillingAccount(acctName, firecloudHasAccount, acct.getDisplayName)
          }
```

Reply:
@marctalbott I tried that, but the Future complicates things.

```
Future.collect partial function is not defined at: false
scala.concurrent.Future$$anon$1: Future.collect partial function is not defined at: false

Reply:
I pushed up the syntax of using `forall`.